runtime/src/iree/modules/hal/debugging.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_MODULES_HAL_DEBUGGING_H_
+#define IREE_MODULES_HAL_DEBUGGING_H_
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+#include "iree/vm/api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+//===----------------------------------------------------------------------===//
+// Debug Sink
+//===----------------------------------------------------------------------===//
+
+// Receives a
+typedef iree_status_t(
+    IREE_API_PTR* iree_hal_module_buffer_view_trace_callback_fn_t)(
+    void* user_data, iree_string_view_t key, iree_host_size_t buffer_view_count,
+    iree_hal_buffer_view_t** buffer_views, iree_allocator_t host_allocator);
```

Comment:
Why is `iree_allocator_t host_allocator` an argument, is it really required there? Can't it be a part of `user_data`?

Reply:
It's the allocator from the HAL module - an implementation could ignore it and use their own in user_data, but this is the only way to get at the HAL module one.